mantidproject/mantid
# Create AlgorithmRunner with no Qt

### Description of work
This PR renames the old `AlgorithmRunner` class to `QtAlgorithmRunner` because it is a QObject and uses qt signals. The PR also creates a new class called `AlgorithmRunner` which does the same as `QtAlgorithmRunner` but without qt signals. The plan is to use the new  `AlgorithmRunner` class in the Indirect/Inelastic interfaces.

*There is no associated issue.*

### To test:
Do some testing of the Reflectometry interface
I have already done testing of the Indirect interfaces
Check the new `AlgorithmRunnerTest` passes

*This does not require release notes* because **it is an internal change**

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
